Source/Plugin.BLE/Windows/BleImplementation.cs

```diff
@@ -21,30 +23,67 @@ public class BleImplementation : BleImplementationBase
 
         private BluetoothLEHelper _bluetoothHelper;
 
+        private Radio? _radio;
+
         protected override IAdapter CreateNativeAdapter()
         {
             return new Adapter(_bluetoothHelper);
         }
 
         protected override BluetoothState GetInitialStateNative()
         {
-            //The only way to get the state of bluetooth through windows is by
-            //getting the radios for a device. This operation is asynchronous
-            //and thus cannot be called in this method. Thus, we are just
-            //returning "On" as long as the BluetoothLEHelper is initialized
             if (_bluetoothHelper == null)
             {
                 return BluetoothState.Unavailable;
             }
-            return BluetoothState.On;
+
+            return _radio is not null
+                ? ToBluetoothState(_radio.State)
+                : BluetoothState.Unknown;
         }
 
         protected override void InitializeNative()
         {
             //create local helper using the app context
             var localHelper = BluetoothLEHelper.Context;
             _bluetoothHelper = localHelper;
+
+            // Wait for the async radio response to catch up, otherwise it will be null
+            // If it falls through, the StateChanged event doesn't get fired in time
+            Task.Run(InitRadioStateAsync).Wait(10);
```

Comment:
I don't quite understand your argumentation regarding the `Wait` here. You changed the timespan from 100ms to 20ms to 10ms, but it's completely unclear how these values are motivated, or why one is better than the other. Wouldn't it be best to wait indefinitely (without a timeout parameter) until the state initialization has fully completed? Or am I missing something?

Reply:
Though I admit to not being a fan of hard timeouts in code, the provided implementation was to get things rolling while delivering a fix.  In the case of 100ms -> 10ms was merely squeezing out the most minimum clock cycles.

Normally in my own systems, I would set an uninitialized state and wait for the event to come back (i.e. `Radio_StateChanged(..)`) to set it accordingly. However, with UWP system having or not having a BLE radio, I'm sure of the effects of doing something like `WaitFof(() => State != BluetoothState.Unknown, MaxTimeoutHere);`

Although, perhaps, this is exactly what we could use. What are your thoughts?

Reply:
Well, my first suggestion would be to change line 53 to ...
```
Task.Run(InitRadioStateAsync).Wait();
```
... in order to make sure that we're waiting for the task to fully run through. If you pass any `timeout` parameter, you always risk that the execution continues before the task has finished and the initialization is not done yet, so that we don't have a well-defined initial state.

Or is there any reason why this would not be a good idea?

Reply:
Made the change & pushed it up. So far, so good. It's been a while since this PR was opened, and I can't reproduce the reason for needing the 10ms.

Do you mind giving this branch a shot on your machine w/ toggling the BLE radio on and off. Just want to make sure that things are good to go here.

Reply:
Sorry, I haven't managed to verify the functionality yet (Windows is really not my native habitat 😉, plus: the sample client currently does not seem to show the radio state).

Is there any easy way to check the changed behavior on your branch? If not, maybe one could extend the sample client to display the radio state in some way?

Have you tested it yourself by now? (Or maybe someone else has?)